### PR TITLE
Dev 74757

### PR DESCRIPTION
**Pull request by htrungcao**
### Description of your changes

Add functionality to retrieve the Catalog asset type ids

---

### JIRA reference

DEV-74757

---

### Impact Analysis

---

#### Checklist

- [x] I have performed a self-review of my code
- [x] My code follows the contribution guidelines of this project
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/helper.py [hunk 1]
```diff
@@ -198,3 +241,18 @@ def validate_inputs():
...
+    with open(path) as f:
+        lineages = json.load(f)
+        for lineage in lineages:
+            for src_trg in lineage:
+                types.add(lineage[src_trg].get("leaf", {}).get("type"))
+                types.add(lineage[src_trg].get("parent", {}).get("type"))
+                for node in lineage[src_trg].get("nodes", []):
+                    types.add(node.get("type"))
+    types.remove(None)
+    f.close()
```
Comment (kristofvancoillie) [inline src/helper.py:257]: I don't think this is needed as you have it in a `with` statement already

File: src/helper.py [hunk 2]
```diff
@@ -198,3 +241,18 @@ def validate_inputs():
 
     validate_inputs()
     return get_assets_fullname_from_collibra()
+
+
+def get_asset_types_name_from_lineage_json_file(path: str) -> set:
+    types = set([])
```
Comment (kristofvancoillie) [inline src/helper.py:247]: why not just `set()` ?

File: src/helper.py [hunk 3]
```diff
@@ -198,3 +241,18 @@ def validate_inputs():
 
     validate_inputs()
     return get_assets_fullname_from_collibra()
+
+
+def get_asset_types_name_from_lineage_json_file(path: str) -> set:
```
Comment (kristofvancoillie) [inline src/helper.py:246]: missing return statement?
would be good to create a small test for this as well by adding a simple lineage.json file in the tests folder?

File: src/helper.py [hunk 4]
```diff
@@ -198,3 +241,18 @@ def validate_inputs():
...
+    types = set([])
+    with open(path) as f:
```
Comment (kristofvancoillie) [inline src/helper.py:248]: should we check if file exist?

File: tools/collect_assets_type.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    with open("metadata.json", "w") as f:
+        json.dump(asset_types, f, indent=4, default=pydantic_encoder)
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument("-c", "--collibraInstance")
+    parser.add_argument("-u", "--username")
+    parser.add_argument("-p", "--password")
+    parser.add_argument("-a", "--applicationName")
```
Comment (kristofvancoillie) [inline tools/collect_assets_type.py:19]: might be good to provide the 'help' argument here, especially for the not obvious ones like applicationName and assetType:

```
parser.add_argument(
        "target_directory", help="Target directory in which the files for batch custom lineage will be stored"
    )
```

File: src/helper.py (hunk 4 above)
Comment (htrungcao) [reply src/helper.py:248]: I did not want to add a try/expect for that, so the user using the function can get the exception.
Do you think we still need a try/expect based on that?

**Review by kristofvancoillie — CHANGES_REQUESTED** [review]
Also don't forgot to update:

- README
- CHANGELOG
- version (in setup.cfg)

**Review by kristofvancoillie — APPROVED** [review]

**Pull request merged**